hydra/conf/__init__.py

```diff
@@ -98,13 +98,18 @@ class HydraConf:
             {"hydra_help": "default"},
             {"hydra_logging": "default"},
             {"job_logging": "default"},
+            {"env": "default"},
+            {"callbacks": "default"},
```

Comment:
env should be last because it should be allowed to override any of the config groups defined above.

Reply:
Also, we can set it to None here and then we wouldn't need env/default.yaml.

Reply:
> env should be last because it should be allowed to override any of the config groups defined above.

Yea agreed env should be the last compared to the other config groups.  But I somehow feel it should be before `callbacks`. If we put this before callbacks, we will get to call our default callbacks before custom ones. We can be sure of what configs we are getting, in case the custom callbacks modifies the GlobalHydra configs etc. 

> Also, we can set it to None here and then we wouldn't need env/default.yaml.

I tried to set this to none here, but as a result, I cannot prepend a `default.yaml` config via the SearchPathPlugin.

Reply:
a specific env file can add some unconditional callbacks for this purpose.

hydra/env/fbcode.yam
```yaml
defaults:
  - hydra/output: fbcode
  - hydra/callbacks/fbcode # not overridable. this does not interact with the choice of the config group hydra/callbacks.
```

However, in some environment you may want to add some default callbacks that the user can override:
```yaml
defaults:
  - hydra/output: fbcode
  - hydra/callbacks: # overrideable
    - foo
    - bar
```

Reply:
I see, thanks, makes sense! I've fixed the order.

Reply:
The new Defaults List is super powerful. It takes a while to really grasp all the things it can do :).